ci/Dockerfile

```diff
@@ -51,29 +52,25 @@ WORKDIR /taichi-dev
 
 # Prevent docker caching when head changes
 ADD https://api.github.com/repos/taichi-dev/taichi/git/refs/heads/master version.json
-RUN git clone https://github.com/taichi-dev/taichi --depth=1 --branch=master
+RUN git clone https://github.com/taichi-dev/taichi --branch=master
+WORKDIR ./taichi
+RUN git checkout $COMMIT_SHA
```

Comment:
Could you do a validation on the SHA to see if its not empty? Otherwise abort with an error message